pytorch_lightning/metrics/converters.py

```diff
@@ -265,23 +279,34 @@ def sync_ddp_if_available(result: Union[torch.Tensor],
 
         if reduce_op is None:
             reduce_op = torch.distributed.ReduceOp.SUM
-        elif isinstance(reduce_op, str) and reduce_op in ('avg', 'mean'):
+        elif isinstance(reduce_op, str) and reduce_op in ("avg", "mean"):
             reduce_op = torch.distributed.ReduceOp.SUM
             divide_by_world_size = True
 
         # sync all processes before reduction
         torch.distributed.barrier(group=group)
-        torch.distributed.all_reduce(result, op=reduce_op, group=group,
-                                     async_op=False)
+        torch.distributed.all_reduce(result, op=reduce_op, group=group, async_op=False)
 
         if divide_by_world_size:
             result = result / torch.distributed.get_world_size(group)
 
     return result
 
+def at_least_1d(tensor: Union[np.ndarray, torch.Tensor]) ->  Union[np.ndarray, torch.Tensor]:
+    """Makes sure the tensor is at least of 1d shape
+
+    Args:
+        tensor: the tensor or array to check the shape for
+
+    Returns:
+        the optionally reshaped tensor
+    """
+    if tensor.shape == ():
+        tensor = tensor.reshape(1,)
+    return tensor
+
 
-def gather_all_tensors_if_available(result: Union[torch.Tensor],
-                                    group: Optional[Any] = None):
+def gather_all_tensors_if_available(result: Union[torch.Tensor], group: Optional[Any] = None):
```

Comment:
Note that this function is has an error and we probably need https://github.com/PyTorchLightning/pytorch-lightning/pull/3319 merged first

Reply:
I saw that one, but I decided to use this for implementation anyways, since I knew it will be fixed :)